packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,251 @@
+import '@aws-cdk/assert/jest';
+import { arrayWith, objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Create a basic canary', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {});
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'canariescanary8f7842',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have specified name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary returns error when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {canaryName: 'myCanary'})).toThrowError('Canary Name must be lowercase, numbers, hyphens, or underscores (no spaces)');
```

Comment:
This lacks spacing and generally speaking isn't super readable

```suggestion
  expect(() => new synthetics.Canary(stack, 'Canary', { canaryName: 'myCanary' }))
    .toThrowError('Canary Name must be lowercase, numbers, hyphens, or underscores (no spaces)');
```

Additionally, I **passionately** hate format validation errors that do not tell you what value failed validating... You have this information at time of making the error message, why would you force me to go dig it up again?